canard.c

```diff
@@ -62,14 +63,34 @@
 struct CanardTxQueueItem
 {
     CanardTxQueueItem* next;
-    CanardCANFrame frame;
+
+    /* Using structs with flexible array members as members in other structs are not technically correct in C.
```

Comment:
@pavel-kirienko @thirtytwobits 

The ignore warning is to not trigger `-Werror` in CI. 

How bad do you think doing this is? It works(tm) with gcc and clang, but technically not allowed in the C specification. The alternative is far from great.